ListDetail/src/androidTest/java/com/microsoft/device/display/samples/listdetail/ListDetailTest.kt

```diff
@@ -0,0 +1,75 @@
+package com.microsoft.device.display.samples.listdetail
+
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.window.testing.layout.WindowLayoutInfoPublisherRule
+import com.microsoft.device.display.samples.listdetail.models.images
+import com.microsoft.device.display.samples.listdetail.ui.theme.ListDetailComposeSampleTheme
+import com.microsoft.device.display.samples.listdetail.ui.view.ListDetailApp
+import com.microsoft.device.dualscreen.testutils.getString
+import com.microsoft.device.dualscreen.testutils.simulateHorizontalFold
+import com.microsoft.device.dualscreen.testutils.simulateVerticalFold
+import com.microsoft.device.dualscreen.windowstate.WindowState
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.rules.TestRule
+
+class ListDetailTest {
+    private val composeTestRule = createAndroidComposeRule<MainActivity>()
+    private val publisherRule = WindowLayoutInfoPublisherRule()
+
+    @get: Rule
+    val testRule: TestRule
+
+    init {
+        testRule = RuleChain.outerRule(publisherRule).around(composeTestRule)
+        RuleChain.outerRule(composeTestRule)
+    }
+
+    @Test
+    fun app_verticalFold_showDetailAfterListClicks() {
+        composeTestRule.setContent {
+            ListDetailComposeSampleTheme() {
+                ListDetailApp(WindowState(hasFold = true, isFoldHorizontal = false))
+            }
+        }
+
+        // Simulate vertical fold
+        publisherRule.simulateVerticalFold(composeTestRule)
+
+        images.forEachIndexed { index, _ ->
+            // Click on list item
+            composeTestRule.onNodeWithTag(
+                index.toString()
+            ).performClick()
+
+            composeTestRule.onNodeWithTag(
+                composeTestRule.getString(R.string.image_tag) + index.toString()
+            ).assertExists()
+        }
+    }
+
+    @Test
+    fun app_horizontalFold_showsList() {
+        composeTestRule.setContent {
+            ListDetailComposeSampleTheme() {
+                ListDetailApp(WindowState(hasFold = true, isFoldHorizontal = true))
+            }
+        }
+
+        // Simulate horizontal fold
+        publisherRule.simulateHorizontalFold(composeTestRule)
+
+        // Assert the list view is not shown
+        composeTestRule.onNodeWithTag(
+            composeTestRule.getString(R.string.list_view)
+        ).assertExists()
+
+        // Assert the detail view is now shown
+        composeTestRule.onNodeWithTag(
+            composeTestRule.getString(R.string.detail_view)
+        ).assertDoesNotExist()
```

Comment:
It looks like the "now" and "not" should be switched, and also I think it would be good to add header comments to both tests in this file!